tsl/src/hypercore/hypercore_handler.c

```diff
@@ -394,6 +395,34 @@ static bool hypercore_getnextslot_noncompressed(HypercoreScanDesc scan, ScanDire
 static bool hypercore_getnextslot_compressed(HypercoreScanDesc scan, ScanDirection direction,
 											 TupleTableSlot *slot);
 
+/*
+ * Skip scanning compressed data in a table scan.
+ *
+ * This function can be called on a scan descriptor to skip scanning of
+ * compressed data. Typically called directly after table_beginscan().
+ */
+void
+hypercore_scan_set_skip_compressed(TableScanDesc scan, bool skip)
+{
+	HypercoreScanDesc hscan;
+
+	if (!REL_IS_HYPERCORE(scan->rs_rd))
+		return;
+
+	hscan = (HypercoreScanDesc) scan;
+
+	if (skip)
+	{
+		scan->rs_flags |= SO_HYPERCORE_SKIP_COMPRESSED;
+		hscan->hs_scan_state = HYPERCORE_SCAN_NON_COMPRESSED;
+	}
+	else
+	{
+		scan->rs_flags &= ~SO_HYPERCORE_SKIP_COMPRESSED;
+		hscan->hs_scan_state = HYPERCORE_SCAN_START;
```

Comment:
Nit: Maybe `...SCAN_START` should be `...SCAN_COMPRESSED` (or let `....SCAN_COMPRESSED` be an alias for `...SCAN_START`) since we conceptually might change the order of the scans if we deem it useful and then we might miss modifying this location, potentially introducing a regression.

Reply:
This is a change outside the scope of this PR since those scan states are not introduced in this PR.